kernel/Kconfig

```diff
@@ -922,6 +922,7 @@ config TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
 config THREAD_LOCAL_STORAGE
 	bool "Thread Local Storage (TLS)"
 	depends on ARCH_HAS_THREAD_LOCAL_STORAGE && TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
+	default y if ARM || RISCV
```

Comment:
FWIW: if you feel you need this to get coverage for TLS, then... we probably don't have sufficient coverage for TLS in our default twister platforms.  Maybe we should pick one or two of the emulation targets and turn this on for them?

Reply:
I think it's pretty clear given the bugs I've found that the small number of TLS tests aren't catching all of the issues. The real adventure is that enabling THREAD_LOCAL_STORAGE now changes core Zephyr functionality, and it's really hard to test that code without exercising a *lot* of potential configurations. From my perspective, enabling THREAD_LOCAL_STORAGE mostly adds code to the system, rarely replacing existing code. As such, maybe we could just turn this on for twister runs on arm/riscv (and other, we we finish TLS support). That way, you'll test all of the new code and only skip a tiny amount of replaced code found in the non-TLS paths. I'm not sure how to configure twister to use a specific configuration addition for all tests that will run though?

Reply:
Yeah, been there.  We have some of the emulation platforms arbitrarily wired to use 32 vs. 64 bit timeouts and/or different scheduler run queue backends just to be sure that we cover all the edges somewhere.  I'd be very happy to see e.g. qemu_riscv32 or whatever be the mine canary for TLS.  We should probably do it at the board level and not arch though.

Reply:
Might be good to pick a QEMU board for each TLS-supporting arch? So much of the TLS code is architecture specific in Zephyr.

Reply:
I've added this stanza to boards/riscv/qemu_riscv32/Kconfig.defconfig which seems to do the trick:
```
# Use thread local storage by default so that
# this feature gets more CI coverage.
config THREAD_LOCAL_STORAGE
	default y
```
Let me know if we want to do anything else in this branch; I believe it's ready to merge now.